src/app/system/account-number-preferences/account-number-preferences.component.ts

```diff
@@ -0,0 +1,64 @@
+/** Angular Imports */
+import { Component, OnInit, ViewChild } from '@angular/core';
+import { MatTableDataSource, MatPaginator, MatSort } from '@angular/material';
+import { ActivatedRoute } from '@angular/router';
+
+/**
+ * Account Number Preferences Component.
+ */
+@Component({
+  selector: 'mifosx-account-number-preferences',
+  templateUrl: './account-number-preferences.component.html',
+  styleUrls: ['./account-number-preferences.component.scss']
+})
+export class AccountNumberPreferencesComponent implements OnInit {
+
+  /** Account Number Preferences data. */
+  accountNumberPreferencesData: any;
+  /** Columns to be displayed in account number preferences table. */
+  displayedColumns: string[] = ['accountType'];
+  /** Data source for account number preferences table. */
+  dataSource: MatTableDataSource<any>;
+
+  /** Paginator for account number preferences table. */
+  @ViewChild(MatPaginator) paginator: MatPaginator;
+  /** Sorter for account number preferences table. */
+  @ViewChild(MatSort) sort: MatSort;
+
+  /**
+   * Retrieves the account number preferences data from `resolve`.
+   * @param {ActivatedRoute} route Activated Route.
+   */
+  constructor(private route: ActivatedRoute) {
+    this.route.data.subscribe((data: { accountNumberPreferences: any }) => {
+      this.accountNumberPreferencesData = data.accountNumberPreferences;
+    });
+  }
+
+  /**
+   * Sets the account number preferences table.
+   */
+  ngOnInit() {
+    this.setAccountNumberPreferences();
+  }
+
+  /**
+   * Initializes the data source, paginator and sorter for account number preferences table.
+   */
+  setAccountNumberPreferences() {
+    this.dataSource = new MatTableDataSource(this.accountNumberPreferencesData);
+    this.dataSource.paginator = this.paginator;
+    this.dataSource.sortingDataAccessor = (accountNumberPreference: any, property: any) => {
+      return accountNumberPreference.accountType.value;
+    };
+    this.dataSource.sort = this.sort;
+  }
+
+  /**
+   * Filters data in account number preferences table based on passed value.
+   * @param {string} filterValue Value to filter data.
+   */
+  applyFilter(filterValue: string) {
+    this.dataSource.filter = filterValue.trim().toLowerCase();
```

Comment:
The filter is not working as the data is in a nested object. You'll have to make a custom filter to make it work.